department-of-veterans-affairs/caseflow
# Pamatya/appeals 45665

Resolves [APPEALS-45665](https://jira.devops.va.gov/browse/APPEALS-45665)

# Description
Please explain the changes you made here.
1. Added Tabs - My saved searches and All saved searches which has table and columns -> 
  a) Search Name, Saved Date, Admin, Description 
  b) Ability to sort Search Name, Saved Date, Admin columns 
2. Creation of mock data to populate table
3. Searches to populate table should be displayed as single-select radio button list
4. if no search are available then it will be an empty table
5.  Default search sort -> Most recent first. 
6.  My saved searches will have searches for all current admin user. 
7. All saved searches will include all the user's search list. 

## Acceptance Criteria
- [ ] Creation of header, description, My saved searches tab, All saved searches tab 
- [ ] My saved searches and All saved searches tab columns: 
          Search Name, Saved Date, Admin, Description 
          Ability to sort Search Name, Saved Date, Admin columns 
- [ ] Creation of mock data to populate table 
- [ ] Searches to populate table should be displayed as single-select radio button list 
- [ ] Default sort top to bottom based on most recent save to earliest 
- [ ] My saved searches tab 
         Include table of saved searches for current user/CSS ID 
- [ ] All saved searches 
          Include table of comprehensive list based on VHA admin group 
          Pagination should include up to 10 searches per page 
## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. Login as a Admin VHA user (VHAADMIN1 or VHAADMIN2) - go to page : http://localhost:3000/decision_reviews/vha/report/searches
3. you should be able to see two tabs with names -> My saved searches and All saved searches populated. 
4. In my saved searches, you should be able to see only records of the users that you are logged in as. 
5. in All saved searches -> you should be able to see all the mock saved search records with pagination.  10 per page limit. 
6. Sorting is by saved date so they will be sorted as recent first. 
7.  Try clicking on the radio button it should be single select meaning no multiple select will be allowed. 
8. To test the empty table view you will have to update the mock data at test/data/nonComp/savedSearchesData.js. You will have to replace `rows` under savedSearches to just be rows: [] `empty array` and save. 


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
